### PR TITLE
add metric bind addr and auth proxy service

### LINKED ISSUE #3225: Consider if to add --metrics-bind-addr=127.0.0.1:8080 in manager.yaml
While working on https://github.com/kubernetes-sigs/cluster-api/pull/6117 we noticed that latest CAPA doesn't have "--metrics-bind-addr=127.0.0.1:8080" in your manager.yaml (see
https://cluster-api.sigs.k8s.io/developer/providers/v0.3-to-v0.4.html#required-kustomize-changes-to-have-a-single-manager-watching-all-namespaces-and-answer-to-webhook-calls).

Applying this change is not mandatory, but FYI deploying Prometheus in local dev env using the CAPI Tiltfile won't work without this change.

### PR DESCRIPTION
**Pull request by Matheusafonsouza**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

- Add metric bind addr on manage yaml
- Add auth proxy service for cluster

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3225

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
None
```

### CODE REVIEW

File: config/rbac/auth_proxy_service.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    control-plane: capa-controller-manager
+  name: controller-manager-metrics-service
+  namespace: system
+spec:
+  ports:
+  - name: https
+    port: 8443
+    targetPort: https
+  selector:
+    control-plane: capa-controller-manager
```
Comment (sedefsavas) [inline config/rbac/auth_proxy_service.yaml:18]:
```suggestion
    control-plane: capa-controller-manager
    
```

**Comment by sedefsavas** [comment]
Thanks for the PR @Matheusafonsouza!

LGTM after minor comment.

/ok-to-test

File: config/rbac/auth_proxy_service.yaml (hunk 1 above)
Comment (Matheusafonsouza) [reply config/rbac/auth_proxy_service.yaml]: Done! @sedefsavas

File: config/rbac/auth_proxy_service.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    prometheus.io/port: "8443"
+    prometheus.io/scheme: https
+    prometheus.io/scrape: "true"
+  labels:
+    control-plane: capa-controller-manager
+  name: controller-manager-metrics-service
```
Comment (sedefsavas) [inline config/rbac/auth_proxy_service.yaml:10]: Rechecking, this should not be part of the manifests we release, it is optional for metrics. 

Let's remove this whole file. My bad.

Comment (Matheusafonsouza) [reply config/rbac/auth_proxy_service.yaml]: Removed!

**Comment by sedefsavas** [comment]
failure was due to irrelevant timeout.



/test pull-cluster-api-provider-aws-test

**Pull request merged**